src/main/java/com/chartboost/sdk/sample/consent/AddConsentDialog.java

```diff
@@ -0,0 +1,175 @@
+        RadioButton coppaRadio = v.findViewById(R.id.consent_dialog_coppa_radio);
+        RadioButton customRadio = v.findViewById(R.id.consent_dialog_custom_radio);
+
+        gdprRadio.setOnClickListener(v1 -> {
+            ccpaRadio.setChecked(false);
+            coppaRadio.setChecked(false);
+            customRadio.setChecked(false);
+            consentType = CONSENT_TYPE.GDPR;
+            switchView();
+        });
+
+        ccpaRadio.setOnClickListener(v1 -> {
+            gdprRadio.setChecked(false);
+            coppaRadio.setChecked(false);
+            customRadio.setChecked(false);
+            consentType = CONSENT_TYPE.CCPA;
+            switchView();
+        });
+
+        coppaRadio.setOnClickListener(v1 -> {
+            gdprRadio.setChecked(false);
+            ccpaRadio.setChecked(false);
+            customRadio.setChecked(false);
+            consentType = CONSENT_TYPE.COPPA;
+            switchView();
+        });
+
+        customRadio.setOnClickListener(v1 -> {
+            gdprRadio.setChecked(false);
+            ccpaRadio.setChecked(false);
+            coppaRadio.setChecked(false);
+            consentType = CONSENT_TYPE.CUSTOM;
+            switchView();
+        });
+
+        createButton.setOnClickListener(v1 -> createConsent());
+        cancelButton.setOnClickListener(v1 -> dismiss());
+        return v;
+    }
+
+    private void createConsent() {
+        String standard = editStandard.getText().toString();
+        String consent = autocompleteConsent.getText().toString();
+        DataUseConsent dataUseConsent = null;
+
+        if (consentType == CONSENT_TYPE.GDPR) {
+            if (GDPR.GDPR_CONSENT.BEHAVIORAL.name().equals(consent)) {
+                dataUseConsent = new GDPR(GDPR.GDPR_CONSENT.BEHAVIORAL);
+            } else if (GDPR.GDPR_CONSENT.NON_BEHAVIORAL.name().equals(consent)) {
+                dataUseConsent = new GDPR(GDPR.GDPR_CONSENT.NON_BEHAVIORAL);
+            }
+        } else if (consentType == CONSENT_TYPE.CCPA) {
+            if (CCPA.CCPA_CONSENT.OPT_IN_SALE.name().equals(consent)) {
+                dataUseConsent = new CCPA(CCPA.CCPA_CONSENT.OPT_IN_SALE);
+            } else if (CCPA.CCPA_CONSENT.OPT_OUT_SALE.name().equals(consent)) {
+                dataUseConsent = new CCPA(CCPA.CCPA_CONSENT.OPT_OUT_SALE);
+            }
+        } else if (consentType == CONSENT_TYPE.COPPA) {
+            if ("true".equals(consent)) {
+                dataUseConsent = new COPPA(true);
+            } else if ("false".equals(consent)) {
+                dataUseConsent = new COPPA(false);
+            }
+        } else {
+            consent = editConsent.getText().toString();
+            dataUseConsent = new Custom(standard, consent);
+        }
+
+        if (dataUseConsent == null) {
+            Toast.makeText(getContext(), "Consent creation failed. Wrong standard or consent name.", Toast.LENGTH_SHORT).show();
+        } else {
```

Comment:
Maybe we can avoid this `else` by dismissing and returning here. Same for the rest of the conditions.